test/test_material.py

```diff
@@ -198,3 +199,33 @@ def test_raises_TypeError_when_E_D_is_not_correct_type():
 
     with pytest.raises(TypeError, match="E_D must be either a float, int or a dict"):
         my_mat.get_E_D()
+
+
+@pytest.mark.parametrize(
+    "input_value",
+    [
+        1.0,
+        1,
+        "coucou",
+        lambda T: 1.0 + T,
+    ],
+)
+def test_raises_TypeError_when_D_is_not_correct_type(input_value):
+    """Test that a TypeError is raised when D is not an fem.Function"""
+
+    with pytest.raises(TypeError, match="D must be of type fem.Function"):
+        F.Material(D=input_value)
+
+
+def test_get_diffusion_coefficient_returns_function_when_given_to_D():
+    """Test that the diffusion coefficient is correctly defined when D is a
+    function"""
+
+    V = fem.functionspace(test_mesh.mesh, ("Lagrange", 1))
+    D = fem.Function(V)
+    D.x.array[:] = 2.0
+
+    my_mat = F.Material(D=D)
+    D_out = my_mat.get_diffusion_coefficient()
+
+    assert isinstance(D_out, fem.Function)
```

Comment:
Could you do the following instead?

```python
assert D_out == D
```
Since in theory they should be the same object.